### PR TITLE
#1704 Shapes Lab: Exception when adding a group of shapes

### LINKED ISSUE #1704: Shapes Lab: Exception when adding a group of shapes
#### Environment (e.g. version of Microsoft PowerPoint)
Unknown

#### Steps to replicate
Add a group of shapes to Shapes Lab

#### Expected behavior
Adds a group of shapes to Shapes Lab

#### Actual behavior

```
System.ArgumentException
Parameter is not valid.

Stack Trace:
   at System.Drawing.Bitmap..ctor(String filename)
   at PowerPointLabs.ShapesLab.LabeledThumbnail.Initialize(String imagePath, String nameLable)
   at PowerPointLabs.ShapesLab.LabeledThumbnail..ctor(String imagePath, String nameLable)
   at PowerPointLabs.ShapesLab.CustomShapePane.AddCustomShape(String shapeName, String shapePath, Boolean immediateEditing)
   at PowerPointLabs.ShapesLab.CustomShapePane.AddShapeFromSelection(Selection selection, ThisAddIn addIn)
   at PowerPointLabs.ActionFramework.ShapesLab.AddShapeActionHandler.ExecuteAction(String ribbonId)
   at PowerPointLabs.ActionFramework.Common.Interface.ActionHandler.Execute(String ribbonId)
````

### PR DESCRIPTION
**Pull request by tankztz**
Fixes #1704 

1. Fixed bugs for Shapes Lab: Exception when adding a group of shapes

2. Fixed bug for ConvertToPicture.cs -> ConvertAndSave(). Before this function execute, the clipboard is necessarily cleared and restored. During this process of clipboard, the Selection objects are cleared outside the function even though it is a local variable. Thus ConvertAndSave() should use selectedShapes instead of selection.

3. Tested in Office 2016

4. Refer to: https://docs.microsoft.com/en-us/previous-versions/office/developer/office-2010/ff762597(v%3doffice.14)

### CODE REVIEW

**Comment by tankztz** [comment]
@junkiattan Ready for review. 

Btw, should I @ you for review matters?

**Comment by tankztz** [comment]
@alexfjw Ready for review.

Btw, should I @ you for review matters?

**Comment by damithc** [comment]
@tankztz sorry we missed this earlier. I'll ping the devs to see who is available.

**Comment by tankztz** [comment]
@damithc  Thank you.

**Comment by leeyh20** [comment]
@tankztz Thank you for your contribution! Sorry for the late reply. Have you been able to reproduce the error on Powerpoint 2016? Where is the clipboard cleared and restored in the code such that it causes this selection error?

From the link you have given us, I read that “The Selection object is deleted whenever you change slides in an active slide view”.  Thus is this error caused by a change of slide or the clipboard?

**Comment by tankztz** [comment]
@leeyh20 
1. I reproduced this issue in Office 365, the bug happens once every time after project started.
2. The clipboard is cleared here [https://github.com/PowerPointLabs/PowerPointLabs/blob/76bf1b77eee0a1430fa085105264f01a3e45b230/PowerPointLabs/PowerPointLabs/Models/PowerPointShapeGalleryPresentation.cs#L124] 
3. This function ConvertAndSave() only have one reference
4. This error is caused by the change of slide, during ConvertAndSave(), the app created a new slide and deleted it
5. To reproduce: I start the program -> open a slides with several rectangle shapes in one page -> select them and click "add group" button

**Comment by tankztz** [comment]
I think this PR can be merged if no further reviews.

File: PowerPointLabs/PowerPointLabs/ShortcutsLab/ConvertToPicture.cs [hunk 1]
```diff
@@ -34,22 +35,17 @@ public static void Convert(PowerPointPresentation pres, PowerPointSlide slide, P
...
-                {
-                    GraphicsUtil.ExportShape(selection.ShapeRange, fileName);
-                }
+                GraphicsUtil.ExportShape(selectedShapes, fileName);
+                return true;
             }
-            else
+            catch
             {
                 MessageBox.Show(ShortcutsLabText.ErrorTypeNotSupported, ShortcutsLabText.ErrorWindowTitle);
```
Comment (leeyh20) [inline PowerPointLabs/PowerPointLabs/ShortcutsLab/ConvertToPicture.cs]: You might want to only show this error "Type is not supported" if` ShapeUtil.IsSelectionShapeOrText(selectedShapes) `is False. Thus if it goes to the `catch` portion, show a more generic error message instead.

**Comment by leeyh20** [comment]
@tankztz We appreciate your contribution, would you be updating this? Thank you!

**Comment by tankztz** [comment]
@leeyh20 Edited code

1. I think the try-catch is good to have 
2. Error message for catch field changed
3. I don't have permission to add label, tested with ppt2016

**Pull request merged**